pypdf/_reader.py

```diff
@@ -1190,6 +1190,9 @@ def _flatten(
         t = "/Pages"
         if PA.TYPE in pages:
             t = pages[PA.TYPE]  # type: ignore
+        # if pdf has no type, considered as a page if /Kids is missing
+        if PA.KIDS not in pages:
+            t = "/Page"
```

Comment:
Is this the same / maybe easier to read?:

```suggestion
        if PA.TYPE in pages:
            t = pages[PA.TYPE]  # type: ignore
        elif PA.KIDS not in pages:
            t = "/Page"
        else:
            t = "/Pages"
```

It might be that this would cause mypy issues

Reply:
Or maybe:

```suggestion
        if PA.TYPE in pages:
            t = pages[PA.TYPE]  # type: ignore
        else:
            t = "/Pages" if PA.KIDS in pages else "/Page"
```

Reply:
The second option is more obvious for me. I propose to keep option 1 just keeping the comment